src/Mod/Arch/archobjects/reinforcement_generic.py

```diff
@@ -0,0 +1,252 @@
+                "App::PropertyPlacementList",
+                "RebarPlacements",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "Placement for each rebar of the reinforcement"
+                )
+            )
+        # TODO: Why ist this property not shown in PropertyEditor
+
+        # BasePlacement
+        if "BasePlacement" not in pl:
+            obj.addProperty(
+                "App::PropertyPlacement",
+                "BasePlacement",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "Rotations of the first rebar in "
+                        "the reinforcement (Yaw-Pitch-Roll)"
+                    )
+                )
+            )
+
+        # Host
+        if "Host" not in pl:
+            obj.addProperty(
+                "App::PropertyLink",
+                "Host",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "The structure object that hosts this rebar"
+                    )
+            )
+
+        # Amount
+        if "Amount" not in pl:
+            obj.addProperty(
+                "App::PropertyInteger",
+                "Amount",
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP("App::Property", ("The amount of rebars")),
+            )
+            obj.setEditorMode("Amount", 1)
+
+        # TotalLength
+        if "TotalLength" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "TotalLength",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property", ("The total length of all rebars")
+                ),
+            )
+            obj.setEditorMode("TotalLength", 1)
+
+    def onDocumentRestored(
+        self,
+        obj
+    ):
+        ArchComponent.Component.onDocumentRestored(self, obj)
+        self.setProperties(obj)
+
+    def execute(
+        self,
+        obj  # why obj? self is the obj?
+    ):
+        if self.clone(obj):
+            return
+        if not obj.BaseRebar:
+            return
+        if not obj.RebarPlacements:
+            return
+        self.build_shape(obj)
+        obj.Amount = len(obj.RebarPlacements)
+        obj.TotalLength = obj.Amount * obj.BaseRebar.Length
+
+        # set Visibility of BaseRebar
+        # this should be done in the Gui Command,
+        # but this dos not yet exist TODO
+        # set view of base rebar to off
+        # if reinforcement shape is not a null shape
+        # TODO may be use another color for base rebar
+        if FreeCAD.GuiUp:
+            if obj.Shape.isNull() is not True:
+                obj.BaseRebar.ViewObject.Visibility = False
+
+    def build_shape(
+        self,
+        obj
+    ):
+        from Part import Shape as sh
```

Comment:
What's the point of this? Just use `Part.Shape()`. The entire `Part` module is already imported.